src/charm.py

```diff
@@ -90,6 +105,151 @@ def _install_snap_packages(self, packages: List[str]) -> None:
                 )
                 raise
 
+    @property
+    def mongos_config(self) -> MongoDBConfiguration:
+        """Generates a MongoDBConfiguration object for mongos in the deployment of MongoDB."""
+        return self._get_mongos_config_for_user(OperatorUser, set(self._unit_ips))
```

Comment:
in the case of subordinate charm, we should connect only to local mongos
```suggestion
        return self._get_mongos_config_for_user(OperatorUser, {"/tmp/mongos.sock"})
```

mongos should be bond to socket `mongos --bind_ip /tmp/mongos.sock`

Reply:
At the risk of sounding naive, why not `mongos --bind_ip localhost`